worlds/pokemon_rb/logic.py

```diff
@@ -33,11 +33,8 @@ def can_flash(state, world, player):
 
 
 def can_learn_hm(state, world, move, player):
-    for pokemon, data in world.local_poke_data.items():
-        if state.has(pokemon, player) and data["tms"][6] & 1 << (["Cut", "Fly", "Surf", "Strength",
-                                                                  "Flash"].index(move) + 2):
-            return True
-    return False
+    hm_mons = world.local_hm_mons_cache[move]
+    return hm_mons and state.has_any(hm_mons, player)
```

Comment:
Is this actually faster than just `state.has_any(hm_mons, player)`?

Reply:
I believe that's why I wrote that, in general it tends to be faster to check than to invoke has_any with an empty list. On the other hand, is it ever the case that there are no mons that can learn an hm? I might not have considered that when I wrote this. Probably better to remove the excess `hm_mons and` then.